error.go

```diff
@@ -0,0 +1,9 @@
+package survey
+
+import (
+	"github.com/AlecAivazis/survey/terminal"
+)
+
+var (
+	InterruptErr = terminal.InterruptErr
```

Comment:
if we are going to define a special error for interrupts (not sure if its necessary tbh) then i would prefer to keep it in a single place, rather than having it spread across both packages.

Reply:
I think it's necessary, since interrupt is commonly treated as a signal of "quit". If the user wants to terminate the program at a survey query, he might send an interrupt signal. The upstream application relied on survey is responsible to react to the interrupt signal properly, whether quit the program or jump out or a loop, etc. Thus InterruptErr should be recognizable from other kinds of errors. How do you think?

Reply:
Yea, i think it makes sense to have a specific error for interrupt that people can check against. Can we just have one in `terminal` and forgo the double declaration?

Reply:
Okay. I may commit later today.